Add CMake build system as a submodule

multiplemonomials opened the pull request:
This PR adds the Amber CMake build system as a submodule, with full build and packaging support.

It also fixes a couple errors I made in my first PR with CMakeLists.txt.

To package using CMake, go to your build directory and run 
```
cmake -DPACKAGE_TYPE=<NSIS or DEB or RPM or Bundle> -DPRINT_PACKAGING_REPORT=TRUE .
```
and then
```
make package
```
All of the relevant options from [the options list on the wiki](http://ambermd.org/pmwiki/pmwiki.php/Main/CMake-Common-Options)

hainm commented:
@Jamie: in the context of Ambertools, I dont feel right if pulling identical submodule for each package (cpptraj, parmed, ...).

Is there any way to make independent cmake stuff for each, and make master cmake in ambertools to reuse them?

hainm commented:
May be add checking if cpptraj inside ambertools, do not use its submodule but using the one in ambertools (as a submodule).

swails commented:
Why don't you feel right about that?

That's the definition of code reuse.

hainm commented:
Uhm, i guess i was wrong.

multiplemonomials commented:
Actually, don't use --recursive when setting up Amber's submodules.  How it works is that during an Amber build, the Amber build system calls into `cphstats/CMakeLists.txt` with a certain variable (`INSIDE_AMBER`) defined.  If it sees this variable, it will assume that the main configuration scripts have already been run, and it won't call into the `cphstats/cmake` folder at all.  Only if it doesn't see `INSIDE_AMBER` will it fall back to its own copy of the CMake scripts.  So you don't need to clone the cmake submodule if it's inside Amber.

hainm commented:
>  Only if it doesn't see INSIDE_AMBER will it fall back to its own copy of the CMake scripts. So you don't need to clone the cmake submodule if it's inside Amber.

the next question is: is it easy to sync?

multiplemonomials commented:
What exactly do you mean by sync? 

Also, I tried to merge my CMake branch with Amber Master, but I keep getting this strange Git error:
```
CONFLICT (modify/delete): AmberTools/src/pytraj deleted in HEAD and modified in 14292c5f72be3eef9c5b56d80aa5e53f8cc26e51. Version 14292c5f72be3eef9c5b56d80aa5e53f8cc26e51 of AmberTools/src/pytraj left in tree at AmberTools/src/pytraj~14292c5f72be3eef9c5b56d80aa5e53f8cc26e51.
Adding AmberTools/src/parmed/setup.py
CONFLICT (modify/delete): AmberTools/src/parmed deleted in HEAD and modified in 14292c5f72be3eef9c5b56d80aa5e53f8cc26e51. Version 14292c5f72be3eef9c5b56d80aa5e53f8cc26e51 of AmberTools/src/parmed left in tree at AmberTools/src/parmed~14292c5f72be3eef9c5b56d80aa5e53f8cc26e51.
Adding AmberTools/src/cpptraj/src/CIFfile.h
CONFLICT (modify/delete): AmberTools/src/cpptraj deleted in HEAD and modified in 14292c5f72be3eef9c5b56d80aa5e53f8cc26e51. Version 14292c5f72be3eef9c5b56d80aa5e53f8cc26e51 of AmberTools/src/cpptraj left in tree at AmberTools/src/cpptraj~14292c5f72be3eef9c5b56d80aa5e53f8cc26e51.
```

I think it's some kind of artifact of merging a submodule-less branch into a submodule branch.  Any idea how to fix it?  There are no files in my work tree like `AmberTools/src/cpptraj~14292c5f72be3eef9c5b56d80aa5e53f8cc26e51`

drroe commented:
Try doing a `git clean -fxd` of `$AMBERHOME/AmberTools/src` first. Make sure there isn't any untracked stuff you want to keep before that.

drroe commented:
You may also have to manually remove conflicting files.

hainm commented:
> What exactly do you mean by sync?

nvm.

multiplemonomials commented:
@swails Could you merge this please?  The CMake-built cphstats won't work properly until this is merged.  Thanks.

multiplemonomials commented:
thanks
